exercises/practice/leap/.approaches/introduction.md

```diff
@@ -0,0 +1,96 @@
+# Introduction
+
+Every fourth year is a leap year (with some exceptions), but let's consider this one condition first. 
+
+To solve the Leap problem, we must determine if a year is evenly divisible by a number or if a reminder of an integer division is zero. 
+Such operation in computing is called [modulo][modulo]. 
+
+Unlike many languages, Elixir does not have [operators][operators] for either integer division or modulo. 
+Instead, it provides [`rem/2`][rem] guard and the [`Integer.mod/2`][mod] function.
+
+The two functions differ in how they work with negative numbers, but since, in this exercise,
+all the numbers are non-negative, both could work, depending on the approach.
+
+## General Solution
+
+To check if a year is divisible by `n`, we can do `rem(year, n) == 0`. We can define a function to make the intent clearer. 
+
+```elixir
+defp divides?(number, divisor), do: rem(number, divisor) == 0 
+```
+
+Any approach to the problem will perform this check three times to check if a year is equally divisible by 4, 100 and 400.
+What will differ is what Elixir features we will use to combine the checks. 
+
+## Approach: Boolean Operators
+
+The full rules are as follows:
+A year is a leap year if 
+* it is divisible by 4 
+* but not divisible by 100
+* unless it is divisible by 400
+
+We can use [boolean operators][boolean-operators] to combine the checks, for example, like so: 
+
+```elixir
+divides?(year, 400) or (not(divides?(year, 100))) and divides?(year, 4)
+```
+In the [boolean operators appraoch][operators-approach] we discuss the details of the solution.
+It includes variations of the operators and their precendence.
+
+## Approach: Function Guards
```

Comment:
This approach should be called 
```suggestion
## Approach: Multiple clause function
```

Guards can be used either in a `case` expression or in multiple clause functions.